src/main/java/fi/aalto/cs/intellij/actions/RequiredPluginsCheckerAction.java

```diff
@@ -0,0 +1,211 @@
+import com.intellij.openapi.updateSettings.impl.PluginDownloader;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Map.Entry;
+import java.util.Objects;
+import java.util.StringJoiner;
+import org.jetbrains.annotations.NotNull;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+//TODO THINK HOW TO TEST & REFACTOR THIS
+public class RequiredPluginsCheckerAction implements StartupActivity {
+
+  private static final Logger logger = LoggerFactory.getLogger(RequiredPluginsCheckerAction.class);
+
+  private static final Map<String, String> requiredPluginNames = new HashMap<>();
+  private static Map<String, String> missingOrDisabledPluginNames = new HashMap<>();
+  private static List<IdeaPluginDescriptor> missingOrDisabledIdeaPluginDescriptors;
+  private static List<IdeaPluginDescriptor> availableIdeaPluginDescriptors;
+
+  /**
+   * The plugin startup activity.
+   */
+  @Override
+  public void runActivity(@NotNull Project project) {
+    populateRequiredPluginNamesMap();
+    filterMissingOrDisabledPluginNames();
+    createListOfMissingOrDisabledPluginDescriptors();
+    checkPluginsStatusAndNotify();
+  }
+
+  /**
+   * Fills the list of required plugin names.
+   *
+   * Later, reading from the from configuration file might occur here.
+   */
+  private void populateRequiredPluginNamesMap() {
+    requiredPluginNames.put("Scala", "org.intellij.scala");
+  }
+
+  /**
+   * Filters out the plugin names that are missing from the current installation.
+   */
+  private void filterMissingOrDisabledPluginNames() {
+    missingOrDisabledPluginNames = requiredPluginNames
+        .entrySet()
+        .stream()
+        .filter(entry -> isPluginMissingOrDisabled(entry.getValue()))
+        .collect(toMap(Entry::getKey, Entry::getValue));
+  }
+
+  /**
+   * Predicate for checking the plugin status.
+   */
+  private static boolean isPluginMissingOrDisabled(String id) {
+    PluginId pluginId = PluginId.getId(id);
+    return !PluginManager.isPluginInstalled(pluginId) || PluginManager
+        .isDisabled(pluginId.getIdString());
+  }
+
+  /**
+   * If there any plugins missing, creates a list of the plugin descriptors for them based on the
+   * publicly available ones.
+   */
+  private void createListOfMissingOrDisabledPluginDescriptors() {
+    if (missingOrDisabledPluginNames.size() > 0) {
+      getAvailablePluginsFromMainRepo();
+      missingOrDisabledIdeaPluginDescriptors = new ArrayList<>();
+      missingOrDisabledPluginNames.forEach((name, id) -> {
+        PluginId missingPluginId = PluginId.getId(id);
+        availableIdeaPluginDescriptors
+            .stream()
+            .filter(
+                availableDescriptor -> availableDescriptor.getPluginId().equals(missingPluginId)
+            )
+            .findFirst()
+            .ifPresent(missingOrDisabledIdeaPluginDescriptors::add);
+      });
```

Comment:
**stream API:** how comfortable are you with Stream API of Java8

Reply:
great stuff!